octoprint_mrbeam/software_update_information.py

```diff
@@ -100,231 +215,244 @@ def switch_software_channel(plugin, channel):
-        # Fix: the develop branches are not formatted as "mrbeam2-{tier}"
-        _b = DEFAULT_REPO_BRANCH_ID[SW_UPDATE_TIER_DEV]
-        kwargs.update(branch=_b, branch_default=_b)
-
-    version = get_version_of_pip_module(package_name, pip_command)
-    if version:
-        kwargs.update(dict(displayVersion=version))
-
-    return _get_package_description(
-        module_id=module_id, tier=tier, pip_command=pip_command, **kwargs
-    )
-
-
-def _get_package_description(
-    module_id,
-    tier,
-    displayName=None,
-    displayVersion=None,
-    type="github_commit",
-    user="mrbeam",
-    repo=None,
-    branch="mrbeam2-{tier}",
-    branch_default="mrbeam2-{tier}",
-    restart="environment",
-    **kwargs
-):
-    """Shorthand to create repo details for octoprint software update plugin to handle."""
-    displayName = displayName or module_id
-    if "{tier}" in branch:
-        branch = branch.format(tier=get_tier_by_id(tier))
-    if "{tier}" in branch_default:
-        branch_default = branch_default.format(tier=get_tier_by_id(tier))
-    update_info = dict(
-        tier=tier,
-        displayName=displayName,
-        displayVersion=displayVersion,
-        user=user,
-        type=type,
-        repo=repo,
-        branch=branch,
-        branch_default=branch_default,
-        restart=restart,
-        **kwargs
-    )
-    return {module_id: update_info}
+        try:
+            _set_octoprint_config(
+                plugin, tier, update_info.pop("octoprint"), beamos_date
+            )
+        except:
+            _logger.error("Error while setting octoprint update config")
+
+        for module_id, module in update_info.items():
+            try:
+                if tier in [
+                    SW_UPDATE_TIER_BETA,
+                    SW_UPDATE_TIER_DEV,
+                    SW_UPDATE_TIER_PROD,
+                ]:
+                    if _is_override_in_settings(
+                        plugin, module_id
+                    ):  # check if update config gets overriden in the settings
+                        return
+                    sw_update_config[module_id] = {}
+                    moduleconfig = sw_update_config[module_id]
+                    moduleconfig.update(defaultsettings)
+
+                    # get update info for tier branch
+                    tierversion = get_tier_by_id(tier)
+
+                    if tierversion in moduleconfig:
+                        module_tier = moduleconfig[tierversion]
+                        _set_update_config_from_dict(
+                            moduleconfig, module_tier
+                        )  # set tier config from default settings
+                    if tierversion in module:
+                        module_tier = module[tierversion]
+                        _set_update_config_from_dict(
+                            moduleconfig, module_tier
+                        )  # override tier config from tiers set in config_file
```

Comment:
the Black formatter does not like comments at the end of lines

Better to have them in a separate line (the previous line) so they don't create these ugly parenthesis-newline patterns